frontend/src/contextProviders/AuthContext.tsx

```diff
@@ -0,0 +1,27 @@
+// import Login from './pages/Login';
+
+import React from 'react';
+import { IAuth, IAuthContext } from '../interfaces';
+import { useLocalStorageState } from '../utils/hooks';
+
+export const AuthContext = React.createContext<IAuthContext>(null!);
+
+export function AuthProvider({ children }: { children: React.ReactNode }) {
+  const [authState, setAuthState] = useLocalStorageState<IAuth | null>('authState', null);
+
+  const redirectToSignIn = () => {
+    window.location.replace(`${import.meta.env.VITE_BACKEND_API_HOST}/v1/auth/google`);
```

Comment:
We can also consider having a backApi.ts file that has an object containing all the backend routes

```
const AUTH = '/auth';
const QUESTION = '/question';
const backendApi = {
  auth: {
    root: AUTH, // doesn't exist. just for example
    google: `${AUTH}/google`
  },
  question: {
    ...
  }
}

Reply:
Will be addressed in separate PR